packages/grafana-data/src/field/fieldDisplay.ts

```diff
@@ -72,6 +73,76 @@ export interface GetFieldDisplayValuesOptions {
 
 export const DEFAULT_FIELD_DISPLAY_VALUES_LIMIT = 25;
 
+interface SparklineHighlightPoint {
+  type: 'point';
+  xIdx: number;
+}
+
+interface SparklineHighlightLine {
+  type: 'line';
+  y: number;
+}
+
+export function getSparklineHighlight(
+  sparkline: FieldSparkline,
+  calc: ReducerID
+): SparklineHighlightPoint | SparklineHighlightLine | void {
+  switch (calc) {
+    case ReducerID.last:
+      return { type: 'point', xIdx: sparkline.y.values.length - 1 };
+    case ReducerID.first:
+      return { type: 'point', xIdx: 0 };
+    case ReducerID.lastNotNull: {
+      for (let k = sparkline.y.values.length - 1; k >= 0; k--) {
+        const v = sparkline.y.values[k];
+        if (v !== null && v !== undefined && !Number.isNaN(v)) {
+          return { type: 'point', xIdx: k };
+        }
+      }
+      return;
+    }
+    case ReducerID.firstNotNull: {
+      for (let k = 0; k < sparkline.y.values.length; k++) {
+        const v = sparkline.y.values[k];
+        if (v !== null && v !== undefined && !Number.isNaN(v)) {
+          return { type: 'point', xIdx: k };
+        }
+      }
+      return;
+    }
+    case ReducerID.min: {
+      let minIdx = -1;
+      let prevMin = Infinity;
+      for (let k = 0; k < sparkline.y.values.length; k++) {
+        const v = sparkline.y.values[k];
+        if (v !== null && v !== undefined && !Number.isNaN(v) && v < prevMin) {
+          prevMin = v;
+          minIdx = k;
+        }
+      }
+      return minIdx >= 0 ? { type: 'point', xIdx: minIdx } : undefined;
+    }
+    case ReducerID.max: {
+      let maxIdx = -1;
+      let prevMax = -Infinity;
+      for (let k = 0; k < sparkline.y.values.length; k++) {
+        const v = sparkline.y.values[k];
+        if (v !== null && v !== undefined && !Number.isNaN(v) && v > prevMax) {
+          prevMax = v;
+          maxIdx = k;
+        }
+      }
+      return maxIdx >= 0 ? { type: 'point', xIdx: maxIdx } : undefined;
+    }
+    case ReducerID.mean:
+      return { type: 'line', y: reduceField({ field: sparkline.y, reducers: [ReducerID.mean] }).mean };
+    case ReducerID.median:
+      return { type: 'line', y: reduceField({ field: sparkline.y, reducers: [ReducerID.median] }).median };
```

Comment:
I already have the logic in for mean and median, and will test it in this PR, but the code to actually use it is commented out below.